### PR TITLE
Fixed issue (#3733) that prevents XR from working in certain circumstances

### LINKED ISSUE #3733: WebXR immersive VR launching issues on Desktop with Oculus AirLink
There's some oddness that I can't explain when launch PlayCanvas WebXR VR experiences on desktop via Oculus AirLink.

Forums thread: https://forum.playcanvas.com/t/recommended-setup-for-xr-development/22794

Setup:
Oculus Quest 1
Connected to Windows 10 with Oculus software using Air link
Using the Oculus OpenXR runtime
Chrome on Windows

PlayCanvas experiences only launches correctly if it's the only tab open that has a WebGL app.

If I have more than one tab (eg and Editor tab and Launch tab), the VR experience fails to load as shown below

https://user-images.githubusercontent.com/16639049/143031177-720c0cae-24cb-479c-a3b6-dbb2440810ef.mp4

However, the hello world from immersive web loads fine no matter the situation: https://immersive-web.github.io/webxr-samples/immersive-vr-session.html

Examples paths I've tried:

Two tabs open with https://immersive-web.github.io/webxr-samples/immersive-vr-session.html , starting VR in either works fine

One tab with https://immersive-web.github.io/webxr-samples/immersive-vr-session.html and another with https://developer.playcanvas.com/en/tutorials/webxr-controllerhand-models/ , starting VR in PlayCanvas fails while immersive-web works fine.

Two tabs open with https://developer.playcanvas.com/en/tutorials/webxr-controllerhand-models/ , neither succeeds entering into VR.

### PR DESCRIPTION
**Pull request by Clicky02**
Fixes #3733 

Fixed bug where having another WebGL instance open in the same browser stops the experience from launching properly.

The problem was solved by preventing playcanvas from rendering new frames when it cannot get the pose of the connected xr device.

It was tested with Oculus Quest 1 and 2 on both Airlink and the link cable.

This is my first PR, so if I need to make any changes let me know.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Hey @Maksims, any comments on this? Looks good to me...

File: src/xr/xr-manager.js [hunk 1]
```diff
@@ -683,6 +685,9 @@ class XrManager extends EventHandler {
         }
 
         const pose = frame.getViewerPose(this._referenceSpace);
+
+        if (!pose) return false;
+
         const lengthNew = pose ? pose.views.length : 0;
```
Comment (mvaligursky) [inline src/xr/xr-manager.js:691]: this line can be simplified now that pose is always valid. It could be
`const lengthNew = pose.views.length;`

**Review by mvaligursky — APPROVED** [review]
It looks great to me, I added one tiny improvement suggestion, but otherwise great job!

File: src/xr/xr-manager.js (hunk 1 above)
Comment (mvaligursky) [inline src/xr/xr-manager.js:691]:
```suggestion
        const lengthNew = pose.views.length;
```

Comment (mvaligursky) [reply src/xr/xr-manager.js]: I've committed the suggested change.

**Pull request merged**